### ISSUE TITLE
Allow to create custom allure steps

### ISSUE THREAD

**Issue by BorisOsipov** [codeceptjs/CodeceptJS#2982]
#### What are you trying to achieve?
As I user I want to create custom steps in allure report

It would be nice to have allure api like

```
const allure = codeceptjs.container.plugins('allure');

allure.step("my awesome step", () => {
 	I.seeElement('.user'); 
	I.seeElement('//button[contains(., "press me")]'); 
})
```
that produces allure step "my awesome step" in the report.

**Comment by Elverslane** [comment]
you could use something like this:
`async selectValuerType() {
        await allure.createStep('Select value type', () => {
            return I.selectOptionByData(RVS.valuerType, value);
        });
    }`

**Comment by BorisOsipov** [comment]
@Elverslane 
1) it is not documented api
2) it is doesnt work - it requires to `await` all calls.

**Comment by Elverslane** [comment]
1. yes
2. it works great despite `await`

**Comment by BorisOsipov** [comment]
2 Nope:

code:
![image](https://user-images.githubusercontent.com/6514276/128839898-8d354a9a-6e3c-4d84-ba4b-d56bf3643aa2.png)

report:
![image](https://user-images.githubusercontent.com/6514276/128839934-4ab28189-474a-4303-b670-dd906166f117.png)

**Comment by BorisOsipov** [comment]
and await all
![image](https://user-images.githubusercontent.com/6514276/128840066-2cd3e067-3222-4d85-8e1d-fa3075b949bf.png)
then you see correct report
![image](https://user-images.githubusercontent.com/6514276/128840100-eaf1c672-8301-4ac9-bea3-f59e9066f10d.png)

**Comment by DavertMik** [comment]
Implemented in #3104
